### PR TITLE
Failure acquiring local host IP address results in 400 #2243

### PR DESCRIPTION
**Pull request by prb112**
- Specific to some setups where the name lookup fails.

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java [hunk 1]
```diff
@@ -44,9 +44,13 @@ private static String determineIP() {
...
             ip = Inet4Address.getLocalHost().getHostAddress();
         } catch (UnknownHostException e) {
-            log.log(Level.SEVERE, "Failure acquiring local host IP address", e);
+            // Defaulting to the localhost, as the machine's name is not registered in the local lookup/dns.
+            log.log(Level.SEVERE, "Failure acquiring local host IP address - be sure the name is registered in the name resolution service", e);
+            ip = "127.0.0.1";
         } catch (NumberFormatException e) {
-            log.log(Level.SEVERE, "Failure parsing local host IP address '" + ip + "'", e);
+            // Defaulting to the localhost, as the machine's name is not registered in the local lookup/dns.
+            log.log(Level.SEVERE, "Failure parsing local host IP address '" + ip + "' - be sure the name is registered in the name resolution service" , e);
```
Comment (lmsurpre) [inline fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java:52]: Won't `ip` always be null here?  Maybe just combine the two catches?

File: fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java [hunk 2]
```diff
@@ -44,9 +44,13 @@ private static String determineIP() {
         try {
             ip = Inet4Address.getLocalHost().getHostAddress();
         } catch (UnknownHostException e) {
-            log.log(Level.SEVERE, "Failure acquiring local host IP address", e);
+            // Defaulting to the localhost, as the machine's name is not registered in the local lookup/dns.
+            log.log(Level.SEVERE, "Failure acquiring local host IP address - be sure the name is registered in the name resolution service", e);
+            ip = "127.0.0.1";
         } catch (NumberFormatException e) {
```
Comment (prb112) [inline fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java:50]:
```suggestion
```

File: fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java [hunk 3]
```diff
@@ -44,9 +44,13 @@ private static String determineIP() {
...
-            log.log(Level.SEVERE, "Failure parsing local host IP address '" + ip + "'", e);
+            // Defaulting to the localhost, as the machine's name is not registered in the local lookup/dns.
```
Comment (prb112) [inline fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java:51]:
```suggestion
```

File: fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java [hunk 4]
```diff
@@ -44,9 +44,13 @@ private static String determineIP() {
         try {
             ip = Inet4Address.getLocalHost().getHostAddress();
         } catch (UnknownHostException e) {
```
Comment (prb112) [inline fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java:46]:
```suggestion
        } catch (UnknownHostException | NumberFormatException e) {
```

File: fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java (hunk 1 above)
Comment (prb112) [reply fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java]: Yes, you are right, I merged the two

File: fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java [hunk 5]
```diff
@@ -43,10 +43,11 @@ private static String determineIP() {
...
             ip = Inet4Address.getLocalHost().getHostAddress();
-        } catch (UnknownHostException e) {
-            log.log(Level.SEVERE, "Failure acquiring local host IP address", e);
-        } catch (NumberFormatException e) {
-            log.log(Level.SEVERE, "Failure parsing local host IP address '" + ip + "'", e);
+        } catch (UnknownHostException | NumberFormatException e) {
+            // Defaulting to the localhost, as the machine's name is not registered in the local lookup/dns.
+            log.log(Level.SEVERE, "Failure acquiring local host IP address - be sure the name is registered in the name resolution service", e);
+            ip = "127.0.0.1";
+            ip = "127.0.0.1";
```
Comment (prb112) [inline fhir-core/src/main/java/com/ibm/fhir/core/util/handler/IPHandler.java:50]:
```suggestion
```

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**